--- origsrc/openwsman-2.6.3/CMakeLists.txt	2016-06-21 06:36:20.000000000 -0500
+++ src/openwsman-2.6.3/CMakeLists.txt	2016-07-03 23:24:46.748447400 -0500
@@ -47,13 +47,13 @@ MESSAGE(STATUS "Building for ${PACKAGE_A
 IF ( DEFINED LIB )
   SET ( LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${LIB}" )
 ELSE ( DEFINED  LIB )
-  IF (APPLE)
+  IF (APPLE OR CYGWIN)
     SET(LIB_SUFFIX "")
-  ELSE (APPLE)
+  ELSE (APPLE OR CYGWIN)
     IF (CMAKE_SIZEOF_VOID_P MATCHES "8")
       SET( LIB_SUFFIX "64" )
     ENDIF(CMAKE_SIZEOF_VOID_P MATCHES "8")
-  ENDIF (APPLE)
+  ENDIF (APPLE OR CYGWIN)
   SET ( LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" )
 ENDIF ( DEFINED  LIB )
 MESSAGE(STATUS "Libraries will be installed in ${LIB_INSTALL_DIR}" )
--- origsrc/openwsman-2.6.3/src/cpp/CMakeLists.txt	2016-06-21 06:36:20.000000000 -0500
+++ src/openwsman-2.6.3/src/cpp/CMakeLists.txt	2016-07-03 23:16:25.044196500 -0500
@@ -9,7 +9,10 @@ SET( wsman_clientpp_LIB_SRCS OpenWsmanCl
 add_library( ${WSMAN_CLIENTPP_PKG} SHARED ${wsman_clientpp_LIB_SRCS})
 
 set_target_properties( ${WSMAN_CLIENTPP_PKG} PROPERTIES VERSION 1.0.0 SOVERSION 1)
-install(TARGETS ${WSMAN_CLIENTPP_PKG} DESTINATION ${LIB_INSTALL_DIR})
+target_link_libraries( ${WSMAN_CLIENTPP_PKG} wsman_client)
+install(TARGETS ${WSMAN_CLIENTPP_PKG} LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+                                      RUNTIME DESTINATION bin
+                                      ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
 
 install(FILES ${wsmaninclude_HEADERS} DESTINATION ${INCLUDE_DIR}/openwsman/cpp)
--- origsrc/openwsman-2.6.3/src/lib/CMakeLists.txt	2016-06-21 06:36:20.000000000 -0500
+++ src/openwsman-2.6.3/src/lib/CMakeLists.txt	2016-07-03 23:15:37.956245400 -0500
@@ -36,7 +36,9 @@ TARGET_LINK_LIBRARIES(wsman ${DL_LIBRARI
 ENDIF( ENABLE_EVENTING_SUPPORT )
 
 SET_TARGET_PROPERTIES(wsman PROPERTIES VERSION 1.0.0 SOVERSION 1)
-INSTALL(TARGETS wsman DESTINATION ${LIB_INSTALL_DIR})
+INSTALL(TARGETS wsman LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+                      RUNTIME DESTINATION bin
+                      ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
 ########### wsman_server ###############
 
